packages/@react-aria/slider/src/useSlider.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {CommonSliderThumbProps, SliderProps} from '@react-types/slider';
+import {computeOffsetToValue} from './utils';
+import {HTMLAttributes, useRef} from 'react';
+import {mergeProps, useDrag1D} from '@react-aria/utils';
+import {SliderState} from '@react-stately/slider';
+import {useLabel} from '@react-aria/label';
+
+interface SliderAria {
+  /** Props for the label element. */
+  labelProps: HTMLAttributes<HTMLElement>,
+
+  /** Props for the root element of the slider component; groups slider inputs. */
+  containerProps: HTMLAttributes<HTMLElement>,
+
+  /** Props for the track element. */
+  trackProps: HTMLAttributes<HTMLElement>,
+
+  /** Partial props that should be passed into `useSliderThumb` for all thumbs.  
+   * Includes the Label ID for labeling slider inputs. */
+  commonThumbProps: CommonSliderThumbProps
```

Comment:
Sounds good.  I called it `commonThumbProps` because it otherwise often interfered with unpacking the result from `useSliderThumb`, like so:

```tsx
const {thumbProps: commonThumbProps} = useSlider(...);
const {thumbProps} = useSliderThumb({ thumbProps: {...commonThumbProps, ...}});
```

But happy to adhere to convention!